pyinfra/api/connectors/winrm.py

```diff
@@ -0,0 +1,188 @@
+from __future__ import print_function, unicode_literals
+
+import click
+import winrm
+
+from pyinfra import logger
+from pyinfra.api.exceptions import ConnectError
+from pyinfra.api.util import make_command, memoize
+
+
+def _raise_connect_error(host, message, data):
+    message = '{0} ({1})'.format(message, data)
+    raise ConnectError(message)
+
+
+@memoize
+# TODO: where should show_warning() be added?
+def show_warning():
+    logger.warning('The @winrm connector is pre-alpha!')
+
+
+def _make_winrm_kwargs(state, host):
+
+    kwargs = {
+    }
+
+    for key, value in (
+        ('username', host.data.winrm_user),
+        ('password', host.data.winrm_password),
+        ('winrm_port', int(host.data.winrm_port or 0)),
+        ('timeout', state.config.CONNECT_TIMEOUT),
+    ):
+        if value:
+            kwargs[key] = value
+
+    # Password auth (boo!)
+    if host.data.winrm_password:
+        kwargs['password'] = host.data.winrm_password
+
+    # TODO: add more auth
```

Comment:
Does winrm have a standard config file (ie SSH's `~/.ssh/config`).

Reply:
I don't think so. We could invent one.